subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -108,7 +108,7 @@ public FileTree getSource() {
      */
     @Deprecated
     @Internal
-    public FileTree getSources() {
+    protected FileCollection getSources() {
         return getProject().files().getAsFileTree();
```

Comment:
Should we create a new one every time here? Should this instead return the same `ImmutableFileCollection` instead?

Reply:
Returning `ImmutableFileCollection.of` now.